modules/gapi/include/opencv2/gapi/cpu/gcpukernel.hpp

```diff
@@ -285,20 +286,22 @@ template<typename Impl, typename... Ins>
 struct OCVSetupHelper<Impl, std::tuple<Ins...>>
 {
     template<int... IIs>
-    static void setup_impl(const GMetaArgs &metaArgs, const GArgs &args, GArg &state,
+    static void setup_impl(const GMetaArgs &metaArgs, const GArgs &args,
+                           const GCompileArgs &compileArgs, GArg &state,
                            detail::Seq<IIs...>)
     {
         // TODO: unique_ptr <-> shared_ptr conversion ?
         // To check: Conversion is possible only if the state which should be passed to
         // 'setup' user callback isn't required to have previous value
         std::shared_ptr<typename Impl::State> stPtr;
-        Impl::setup(detail::get_in_meta<Ins>(metaArgs, args, IIs)..., stPtr);
+        Impl::setup(detail::get_in_meta<Ins>(metaArgs, args, IIs)..., compileArgs, stPtr);
         state = GArg(stPtr);
     }
 
-    static void setup(const GMetaArgs &metaArgs, const GArgs &args, GArg& state)
+    static void setup(const GMetaArgs &metaArgs, const GArgs &args,
+                      const GCompileArgs &compileArgs, GArg& state)
     {
-        setup_impl(metaArgs, args, state,
+        setup_impl(metaArgs, args, compileArgs, state,
```

Comment:
...with Context, we could leave all these methods unmodified.

Reply:
Hopefully we'll not extend it anymore.

Reply:
Got you